Assets/Mirror/Components/NetworkAnimator.cs

```diff
@@ -534,5 +540,75 @@ void RpcOnAnimationResetTriggerClientMessage(int hash)
         }
 
         #endregion
+
+
+        #region intial setup
+        [Server]
+        void SendSetupMessage()
+        {
+            // mark ready on server
+            hasSetup = true;
+
+            using (PooledNetworkWriter writer = NetworkWriterPool.GetWriter())
```

Comment:
Why all this code to do what ClientRpc can do with 2 parameters?

Reply:
This code was moved from `OnSerialize`

On the server side it needs to write the state of each layer and all the parameters. If we were to use ClientRpc then they would have to write their values to arrays which would require a buffer to write to. Instead we can just use PooledNetworkWriter  as it is already a buffer.

On the client side, the Animator might not have been set when the message is received so the message needs to be stored and handled when the animator is set